browser/extensions/extensibles/test/browser/browser_privatetab.js

```diff
@@ -1,3 +1,17 @@
+"use strict";
+
+requestLongerTimeout(2);
+
+async function togglePrivate(tab, skip = false) {
+  if (skip) {
+    return PrivateTab.togglePrivate(window, tab);
+  }
+  await openTabContextMenu(tab);
+  let openPrivate = document.getElementById("toggleTabPrivateState");
+  openPrivate.click();
+  return gBrowser.selectedTab;
```

Comment:
Question: is the gBrowser.selectedTab updated synchronously? If not, then is this the reason for requesting a longer timeout?

Reply:
Sorry, I phrased the question poorly. I was wondering if openPrivate.click() causes gBrowser.selectedTab to be updated synchronously - can we be sure it is updated to the new tab as seems to be the intention?

Reply:
Ah gotcha, in this case yes - gBrowser.selectedTab is only evaluated after the click is registered.